### PR TITLE
Changed "exact match boost" implementation to use a match_phrase query in should

### LINKED ISSUE #4960: Change "exact match boost" implementation to use a `match_phrase` query in `should`
## Problem

We currently use an additional simple query string entry into the boolean filter for search to boost exact phrasal matches on the title of a work:

https://github.com/WordPress/openverse/blob/50e401f17b9290cd373bdced98babd457700b990/api/api/controllers/search_controller.py#L305-L312

## Description

We can bypass simple query string altogether for this, saving some parsing cycles in Elasticsearch, and making our intention much clearer in code (and in the generated query), by using [`match_phrase`](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query-phrase.html) directly. This also obviates the need to mock out the simple-query-string syntax for phrasal matches.

### PR DESCRIPTION
**Pull request by dryruffian**
## Fixes

Fixes #4960  by @sarayourfriend 

## Description

This pull request changes the usage of `simple_query_string` to `match_phrase` as suggested by @sarayourfriend. This will bypass simple query string altogether for this, saving some parsing cycles in Elasticsearch. It will also make our intention much clearer in code (and in the generated query) by using `match_phrase` directly.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (`ov just catalog/generate-docs` for catalog
      PRs) or the media properties generator (`ov just catalog/generate-docs media-props`
      for the catalog or `ov just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by dryruffian** [comment]
hi @sarayourfriend, 
Can you please take a look at my PR. Passed all the API test on my machine. There is some formatting can you guide me through that.

**Comment by sarayourfriend** [comment]
Sure! You should run `ov just lint` on the latest commit of the branch and then commit any changes that appear. Typically we rely on the git pre-commit hook to run the linter for us before we commit changes. Are your git hooks set up? You can check by running `cat .git/hooks/pre-commit` and confirm it looks like this:

```bash
#! /usr/bin/env bash

export TERM=xterm

./ov hook pre-commit "$@"
```

`ov init` from our getting started documentation should have installed the hooks for you. You can run `ov just install-hooks` to re-run `ov`'s hook installation.

**Comment by dryruffian** [comment]
Hey @sarayourfriend, thanks for your help getting this PR to work. I really appreciate it. Can you please take a look at the PR and suggest any changes that might be needed?

File: api/test/unit/controllers/test_search_controller_search_query.py [hunk 1]
```diff
@@ -129,11 +129,11 @@ def test_create_search_query_q_search_with_quotes_adds_raw_suffix(
...
         "should": [
             {
-                "simple_query_string": {
-                    "boost": 10000,
-                    "fields": ["title"],
-                    "query": "The cutest cat",
-                    "flags": DEFAULT_SQS_FLAGS,
+                "match_phrase": {
+                    "title": {
+                        "query": '"The cutest cat"',
```
Comment (sarayourfriend) [inline api/test/unit/controllers/test_search_controller_search_query.py:134]: For other reviewers who notice this double-quoting. It's preserved from the input string. It shouldn't be a problem, it seems that (at least with our current analyser configuration for title) `match_phrase` filters our non-alphanumeric tokens in search. For example, a query for `"house cat photo"` (with the quotes) ends up as this, for the `match_phrase` part of the scoring, and is identical if I drop the quotes in the query string.

```json
{
  "value": 10000,
  "description": "weight(title:\"hous cat photo\" in 95) [PerFieldSimilarity], result of:",
  "details": [
    {
      "value": 10000,
      "description": "score(BooleanWeight), computed from:",
      "details": [
        {
          "value": 10000,
          "description": "boost, query boost",
          "details": []
        }
      ]
    }
  ]
}
```

**Review by sarayourfriend — APPROVED** [review]
Thanks again @dryruffian!

**Pull request merged**